mindsdb/integrations/handlers/neuralforecast_handler/neuralforecast_handler.py

```diff
@@ -0,0 +1,99 @@
+import os
+import string
+import random
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.integrations.handlers.statsforecast_handler.statsforecast_handler import (
+    infer_frequency,
+    transform_to_nixtla_df,
+    get_results_from_nixtla_df,
+)
+from neuralforecast import NeuralForecast
+from neuralforecast.models import NHITS
+from neuralforecast.auto import AutoNHITS
+from ray.tune.search.hyperopt import HyperOptSearch
+
+DEFAULT_FREQUENCY = "D"
+DEFAULT_MODEL_NAME = "NHITS"
+DEFAULT_TRIALS = 20
+MIN_TRIALS_FOR_AUTOTUNE = 3
+
+
+def choose_model(num_trials, horizon, window, exog_vars=[], threshold=MIN_TRIALS_FOR_AUTOTUNE):
+    """Chooses model based on the number of trials allowed by the user.
+
+    A lower args for training time reduces the number of trial models we can
+    search through. Below a certain threshold, we switch to the default
+    NHITS implementation instead of using AutoML to search for the best config.
+    """
+    if num_trials >= threshold:
+        model = AutoNHITS(horizon, gpus=0, num_samples=num_trials, search_alg=HyperOptSearch())
```

Comment:
Any chance that the final window picked here would be larger than the one specified by the user? If so, there's a risk of having missing context when generating predictions, as the mindsdb executor wouldn't know the window is actually larger.

Reply:
Here's what should happen:

1) AutoNHITS will find the best training window as part of its hyperparam search.
2) The trained model will make predictions using this window, it can do this regardless of the length of the df that is supplied through the prediction query.

There's a related issue that users are asked to supply a "Window" arg that's only actually used if MindsDB picks NHITS rather than AutoNHITS. Perhaps we should clarify that in the documentation?

Reply:
EDIT: have now clarified this in the README file.

Reply:
> can do this regardless of the length of the df that is supplied through the prediction query

Makes sense given that the `forecast()` method predicts for the very next out-of-sample timestamps.

I do wonder what'd happen with predictions that are further away (e.g. if horizon is 12, and you provide a DF that starts at `t+13`, would that happen then?